### PR TITLE
Migrate Packaging Builders to .ci.yaml

### PR DESCRIPTION
**Pull request by sealesj**
In order to avoid duplicated infra configurations in flutter and dart-internal due to BCID work on release artifacts, [packaging builder configurations from here](https://cs.opensource.google/flutter/infra/+/main:config/packaging_config.star;l=164-216) are getting moved to this .ci.yaml file.

Issue addressed: https://github.com/flutter/flutter/issues/100806

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -4620,33 +4620,45 @@ targets:
...
     scheduler: release
     bringup: true
     enabled_branches:
       - beta
+    properties:
+      task_name: flutter_packaging_beta_mac
+
 
-  - name: Mac_arm64 flutter_packaging_beta
+  - name: Mac_arm64 flutter_packaging_beta_mac_arm64
```
Comment (godofredoc) [inline .ci.yaml:4643]: Do we need the platform_architecture at the end, we can probably pass a cpu dimension like in here: https://cs.opensource.google/flutter/flutter/+/master:.ci.yaml;l=2023?q=drone%20dimensions&ss=flutter&start=21

Comment (godofredoc) [reply .ci.yaml:4643]: Let's remove the _linux and _mac at the end of the names

Comment (sealesj) [reply .ci.yaml]: Removed

**Review by godofredoc — APPROVED** [review]

File: .ci.yaml (hunk 1 above)
Comment (sealesj) [reply .ci.yaml]: Thank you!

**Comment by sealesj** [comment]
> > auto label is removed for flutter/flutter, pr: 115569, due to - The status or check suite [Linux test_ownership](https://ci.chromium.org/b/8795946278804630753) has failed. Please fix the issues identified (or deflake) before re-applying this label.
> 
> Seems like we also need to update TESTOWNERS

Yes I'm working to figure out what the problem is. I added  `task_name` and `tags` for each target but am still seeing that the tests are parsed as `none` for the owner. I'm looking into the script that is used to parse the yaml now to see what is missing

File: .ci.yaml [hunk 2]
```diff
@@ -4620,6 +4620,58 @@ targets:
...
 
+  - name: Linux flutter_packaging_beta
+    recipe: packaging_v2/packaging_v2
+    timeout: 60
+    scheduler: release
+    bringup: true
+    enabled_branches:
+      - beta
+    properties:
+      task_name: flutter_packaging_beta
```
Comment (yusuf-goog) [inline .ci.yaml:4631]: Should these names be postfixed with OS?

Comment (godofredoc) [reply .ci.yaml:4631]: Do we do something with postfixes?

Comment (yusuf-goog) [reply .ci.yaml:4631]: I don't think so, but the other names in the file are setup in that convention, e.g. line 4612

Comment (godofredoc) [reply .ci.yaml:4631]: aaah, no, many of the values we are providing here are not needed but not sure if removing them will make TESTONWERS unhappy.

**Pull request merged**